Lib/fontTools/varLib/instancer/__init__.py

```diff
@@ -250,6 +252,50 @@ def __post_init__(self):
                 f"minimum={self.minimum:g}, default={self.default:g}, maximum={self.maximum:g})"
             )
 
+    def reverse_negate(self):
+        v = self
+        return self.__class__(-v[2], -v[1], -v[0], v[4], v[3])
+
+    def normalizeValue(self, v, extrapolate=True):
+        lower, default, upper, distanceNegative, distancePositive = self
+        assert lower <= default <= upper
+
+        if not extrapolate:
+            v = max(lower, min(upper, v))
+
+        if v == default:
+            return 0
+
+        if default < 0:
+            return -self.reverse_negate().normalizeValue(-v, extrapolate=extrapolate)
+
+        # default >= 0 and v != default
+
+        if v > default:
+            return (v - default) / (upper - default)
+
+        # v < default
+
+        if lower >= 0:
+            return (v - default) / (default - lower)
+
+        # lower < 0 and v < default
+
+        totalDistance = distanceNegative * -lower + distancePositive * default
+
+        if v >= 0:
+            vDistance = (default - v) * distancePositive
+        else:
+            vDistance = -v * distanceNegative + distancePositive * default
+
+        if totalDistance == 0:
+            # This happens
```

Comment:
when does this situation happen specifically?
IIUC, totalDistance == 0 only when both distanceNegative and distancePositive are 0, which seems unlikely to me in real-world because it would mean fvar axis min==default==max hence the two distances from default zero out.

Reply:
Ah. Now I see it was happening when I was intentionally passing 0,0 as distances in featureVars.py. I removed it, but am not confident it's correct. Lemme think.

Reply:
I believe it's correct.